### ISSUE TITLE
Dynamo user inputs implemented into the Batch Processor

### ISSUE THREAD

**Issue by gnye4598** [bvn-architecture/RevitBatchProcessor#60]
I have several inputs within a Dynamo script that I would typically utilize data-shapes for. It provides a drop-down menu where users can select different inputs before running the Dynamo script. Now that I am using Revit Batch Processor, the drop-down menu appears after each revit file that opens. Is there a way I can batch the inputs of my dynamo script without having to enter in the data each time a Revit document opens?

**Comment by DanRumery** [comment]
Hi @gnye4598 

I think that would involve significant knowledge of the inner workings of Dynamo to add that kind of functionality (but I agree that would be really useful). I wonder if you can setup your workspace so you have a boolean that toggles between the UI-based user input and non-UI-based input (queries / selections made programmatically)

**Comment by gnye4598** [comment]
Hi Dan thanks for the response! Yes I think it does sound like it would need to be played with more specifically in Dynamo. I’m wondering if there’s a way I can have it so the first time a revit document is opened it asks for user inputs and then it stores that data in say an excel sheet. Then each time a revit document opens up after the first one that it references that excel sheet rather than the ui pop up. I bet that could be worked out with a few conditional statements...I’ll keep you up to date on that!

**Comment by gnye4598** [comment]
I shocked myself a little bit and got it to work. I was able to tell dynamo to open a blank excel document, determine that if it is blank it opens a data-shapes input box which asks the user to input data and along with performing the task, saves that new info to excel. So this way when the next revit document is opened instead of opening an empty excel sheet it opens the data that was just manually inserted and uses that instead of opening a data-shapes box. The only issue now is that the Batch Processor attempts to fill in the data-shapes on its own without allowing any user interaction. That might be something that only is the case for the input of a directory path though. I might look for other means of user interface popup boxes.
